site/cds_rdm/inspire_harvester/reader.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 CERN.
+#
+# CDS-RDM is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""Reader component."""
+from urllib.parse import urlencode
+
+import requests
+from datetime import datetime
+from invenio_vocabularies.datastreams.readers import BaseReader
+from invenio_vocabularies.datastreams.errors import ReaderError
+
+
+class InspireHTTPReader(BaseReader):
+    """INSPIRE HTTP Reader."""
+
+    def __init__(self, origin=None, mode="r", since=None, until=None, on=None, inspire_id=None, *args, **kwargs):
+        """Constructor."""
+        self._since = since
+        self._until = until
+        self._on = on
+        self._inspire_id = inspire_id
+        super().__init__(origin, mode, *args, **kwargs)
+
+    def _validate_date_format(self, date):
```

Comment:
isn't date validated by the args schema? Do we need an additional validator?

Reply:
No it is actually not validating the input data, so we need to add the manual validation